File: preorder_tree_traversal.java
/**
 * Definition for binary tree
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public List<Integer> preorderTraversal(TreeNode root) {
        List<Integer> mylist = new ArrayList<Integer>();
        traverse(root, mylist);
        return  mylist;
    }
    
    public void traverse(TreeNode node, List<Integer> mylist){
        if (node == null){return;}
        mylist.add(node.val);
        traverse(node.left,mylist);
        traverse(node.right,mylist);
        
    }
}

File: preorder_tree_traversal.py
# Definition for a  binary tree node
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    # @param root, a tree node
    # @return a list of integers
    def preorderTraversal(self, root):
        node =root
        result =[]
        self.preorder(node, result)
        return result
        
    def preorder(self, node, result):
        if node:
            result.append(node.val)
            self.preorder(node.left,result)
            self.preorder(node.right,result)
    
            
        